src/dialogs/prompt/html.js

```diff
@@ -0,0 +1,122 @@
+const { IS_MAC, IPFS_LOGO_URI } = require('../../common/consts')
+const { styles } = require('./styles')
+
+/**
+ * Generates an HTML string with the given button labels.
+ * The order of the buttons is inverted on macOS to match the behavior
+ * of the OS.
+ *
+ * @param {string[]} labels
+ * @returns {string}
+ */
+function getButtonComponentsHtml (labels) {
+  const buttons = labels.map((txt, i) => `<button ${i === 0 ? 'class="default"' : ''} id="${i}">${txt}</button>`)
+
+  if (IS_MAC) {
+    buttons.reverse()
+  }
+
+  return buttons.join('\n')
+}
+
+/**
+ * @typedef InputConfiguration
+ * @type {object}
+ * @property {string} type
+ * @property {string} name
+ * @property {string} defaultValue
+ * @property {string | null} label
+ * @property {string[] | null} labels
+ */
+
+/**
+ * Generates an HTML string with the given configurations.
+ *
+ * @param {InputConfiguration[]} inputs
+ * @returns {string}
+ */
+function getInputComponentsHtml (inputs) {
+  return inputs.map(({ type, name, label, defaultValue, labels = {} }) => {
+    let str = '<div>'
+
+    switch (type) {
+      case 'checkbox':
+        str += '<div class="inline">'
+        str += `<input type="checkbox" name="${name}" id="${name}" ${defaultValue} />`
+        str += `<label for="${name}">${label}</label>`
+        str += '</div>'
+        break
+      case 'radio':
+        str += '<div class="group">'
+        for (const key in labels) {
+          str += '<div class="inline">'
+          str += `<input type="radio" name="${name}" id="${key}" value="${key}" ${defaultValue === key ? 'checked' : ''} />`
+          str += `<label for="${key}">${labels[key]}</label>`
+          str += '</div>'
+        }
+        str += '</div>'
+
+        break
+      case 'text':
+        str += `<input type="text" name="${name}" id="${name}" value="${defaultValue}" />`
+    }
+
+    str += '</div>'
+    return str
+  }).join('\n')
+}
+
+/**
+ * Generates a base64 encoded URI with the HTML content of a prompt window.
```

Comment:
Why do we need to do this as a base64 encoded URI?

Reply:
We don't need, but it seems to be the easier solution. The other solution is to start a server with an HTML file and somehow costumize the page on the fly. This seems easier for a simple prompt.